crates/common/src/selectors.rs

```diff
@@ -142,19 +142,51 @@ impl SignEthClient {
         selector: &str,
         selector_type: SelectorType,
     ) -> eyre::Result<Vec<String>> {
+        self.decode_selectors(&[selector], selector_type)
+            .await?
+            .pop() // Not returning on the previous line ensures a vector with exactly 1 element
+            .unwrap()
+            .ok_or(eyre::eyre!("No signature found"))
+    }
+
+    /// Decodes the given function or event selectors using https://api.openchain.xyz
+    pub async fn decode_selectors(
+        &self,
+        selectors: &[impl ToString],
+        selector_type: SelectorType,
+    ) -> eyre::Result<Vec<Option<Vec<String>>>> {
+        if selectors.is_empty() {
+            return Ok(vec![]);
+        }
+
         // exit early if spurious connection
         self.ensure_not_spurious()?;
 
+        let selectors: Vec<String> = selectors
+            .iter()
+            .map(|s| s.to_string())
+            .map(|s| if s.starts_with("0x") { s } else { format!("0x{s}") })
+            .collect();
+
+        let expected_len = match selector_type {
+            SelectorType::Function => 10, // 0x + hex(4bytes)
+            SelectorType::Event => 66,    // 0x + hex(32bytes)
+        };
+        if let Some(s) = selectors.iter().find(|s| s.len() != expected_len) {
+            eyre::bail!(
+                "Invalid selector {s}: expected {expected_len} characters (including 0x prefix)."
+            )
+        }
+
         #[derive(Deserialize)]
         struct Decoded {
             name: String,
-            filtered: bool,
```

Comment:
`filter=true` by default ([source](https://docs.openchain.xyz/#/default/get_signature_database_v1_lookup)), so results are already filtered on API side and this field is not in response.